tests/performance_tests/test_memory_usage.py

```diff
@@ -39,8 +40,8 @@ def poly_template(monkeypatch):
     yield folder
 
 
-@pytest.mark.flaky(reruns=5)
-@pytest.mark.limit_memory("130 MB")
+# @pytest.mark.flaky(reruns=5)
+@pytest.mark.limit_memory("1 GB")
```

Comment:
Is it possible to extract the part that takes up so much memory, so the test can have a lower requirement? It is trying to test the memory usage of the update step, but think that will drown in other requirements if it is set to 1GB?

Reply:
I'm not sure, one way to "force" it to use less memory would be to pass through a chunking to the xarray when it opens up the response and does the combination, though this would also slow things down a bit. The reason for the memory usage is mainly that it goes from looping over each single observation, to combining it all at once "in the update step" as the update step will fetch the measured data, which will trigger the (ideally done only once per ensemble) computation.